app/src/main/java/comcmput301f16t01/github/carrier/MainActivity.java

```diff
@@ -164,11 +165,15 @@ public void onPageScrollStateChanged(int state) {
     protected void onResume() {
         super.onResume();
 
-        // Inform the user of unread notifications.
+        // Will toast if there are unread notifications.
+        final Context context = this;
         NotificationController nc = new NotificationController();
-        if (nc.unreadNotification( UserController.getLoggedInUser() )) {
-            promptViewNotifications();
-        }
+        nc.asyncUnreadNotification(UserController.getLoggedInUser(), new Listener() {
+            @Override
+            public void update() {
+                Toast.makeText( context, "You have unread notifications!", Toast.LENGTH_LONG ).show();
```

Comment:
This is too long. Maybe make it Toast.LENGTH_SHORT (or do your own thing to make it somewhere in the middle).